packages/@ama-sdk/core/src/fwk/clients/api-fetch-client.ts

```diff
@@ -141,21 +143,20 @@ export class ApiFetchClient implements ApiClient {
       if (e instanceof CanceledCallError) {
         exception = e;
       } else {
-        exception = new EmptyResponseError(e.message || 'Fail to Fetch', undefined, { apiName, operationId, url, origin });
+        exception = new EmptyResponseError(e.message || 'Fail to Fetch', undefined, {apiName, operationId, url, origin});
       }
     }
 
     try {
       root = body ? JSON.parse(body) : undefined;
     } catch (e: any) {
-      exception = new ResponseJSONParseError(e.message || 'Fail to parse response body', response && response.status || 0, body, { apiName, operationId, url, origin });
+      exception = new ResponseJSONParseError(e.message || 'Fail to parse response body', response && response.status || 0, body, {apiName, operationId, url, origin});
     }
-
     const replyPlugins = this.options.replyPlugins ?
       this.options.replyPlugins.map((plugin) => plugin.load<T>({
         dictionaries: root && root.dictionaries,
         response,
-        reviver,
+        reviver: typeof reviver === 'function' || typeof reviver === 'undefined' ? reviver : (response?.status && reviver[response.status]) || undefined,
```

Comment:
not sure the `|| undefined1 bring something, but OK :D

Reply:
![image](https://github.com/AmadeusITGroup/otter/assets/110408057/2228b91f-fc6c-46d5-957c-b83a66acaf17)

Reply:
I missed the `0` indead. TS is good !